editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -590,10 +431,163 @@ export const previewStore: StateCreator<
+    if (Array.isArray(passportValues) && passportValues.length > 0) {
+      // Check if the existing passport value(s) startsWith at least one option's val (eg passport retains most granular values only)
+      const matchingPassportValues = passportValues.filter((passportValue: any) =>
+        sortedOptions.some((option) =>
+          passportValue?.startsWith(option.data?.val),
+        ),
+      );
+
+      if (matchingPassportValues.length > 0) {
+        let foundExactMatch = false;
+        sortedOptions.forEach((option) => {
+          passportValues.forEach((passportValue: any) => {
+            // An option can be auto-answered if it has direct match in the passport
+            //   or if the passport has a more granular version of the option (eg option is `fruit`, passport has `fruit.apple`) 
+            //     but only in cases where we don't also have the exact match
+            if (passportValue === option.data?.val) {
+              if (option.id) optionsThatCanBeAutoAnswered.push(option.id);
+              foundExactMatch = true;
+            } else if (passportValue.startsWith(option.data?.val) && !foundExactMatch) {
+              if (option.id) optionsThatCanBeAutoAnswered.push(option.id);
+            }
+          });
+        });
+      } else {
+        if (blankOption?.id) optionsThatCanBeAutoAnswered.push(blankOption.id);
+      }
+    } else {
+      // If we don't have any existing passport values for this fn but we do have a blank option,
+      //  proceed through the blank if every option's val has been visited before
+      const sortedOptionVals: string[] = sortedOptions.map((option) => option.data?.val);
+      let visitedOptionVals: string[] = [];
+      visitedFns.forEach(([nodeId, _breadcrumb]) => {
+        flow[nodeId].edges?.map((edgeId) => {
+          if (flow[edgeId].type === TYPES.Answer && flow[edgeId].data?.val) {
+            visitedOptionVals.push(flow[edgeId].data.val);
+          }
+        })
+      });
+
+      // Planning Constraints use a bespoke "_nots" data structure to describe all option vals returned via GIS API
+      //   Concat these onto other visitedOptionVals so that questions about constraints we haven't fetched are put to user exactly once
+      if (visitedFns.some(([nodeId, _breadcrumb]) => flow[nodeId].type === TYPES.PlanningConstraints)) {
+        const nots: string[] | undefined = computePassport()?.data?.["_nots"]?.[data.fn];
+        if (nots) visitedOptionVals = visitedOptionVals.concat(nots);
+      }
+
+      const hasVisitedEveryOption = sortedOptionVals.every(value => visitedOptionVals.includes(value));
+      if (blankOption?.id && hasVisitedEveryOption) optionsThatCanBeAutoAnswered.push(blankOption.id);
+    }
+
+    // Questions 'select one' and therefore can only auto-answer the single left-most matching option
+    if (type === TYPES.Question) {
+      optionsThatCanBeAutoAnswered = optionsThatCanBeAutoAnswered.slice(0, 1);
+    }
```

Comment:
My argument for here rather than within `useEffect` of the component itself is that it's nice to be able to clearly test length of returned array in unit tests for Questions ! Eg test cases where I know that there are _many_ matching paths, but I want to confirm that I'm only going to auto-answer the most-granular, left-most option if my node is a Question